### PR TITLE
Feature Generator Example

### PR DESCRIPTION
**Pull request by minnce**
### Description
We are ensuring that our feature generator behaves as intended

### Testing
Unit tests

### Impact
Nothing specifically impacted, just testing base feature gen

### CODE REVIEW

File: src/pysrc/signal/example_feature_generator.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from pysrc.adapters.messages import SnapshotMessage, TradeMessage
+from pysrc.util.types import Asset
+from pysrc.signal.base_feature_generator import BaseFeatureGenerator
+
+
+class ExampleFeatureGenerator(BaseFeatureGenerator):
+    orderFeatures = ["open", "high", "low", "close"]
```
Comment (echavemann) [inline src/pysrc/signal/example_feature_generator.py:7]: snake

File: src/pysrc/test/unit/signal/test_example_feature_generator.py
Comment (echavemann) [inline src/pysrc/test/unit/signal/test_example_feature_generator.py:1]: is there any way we can pull stuff into higher scope to reduce code duplication?

File: src/pysrc/test/unit/adapters/kraken/future/test_kraken_future_client.py
Comment (echavemann) [inline src/pysrc/test/unit/adapters/kraken/future/test_kraken_future_client.py:1]: ?????????????????

File: poetry.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.8.4 and should not be changed by hand.
```
Comment (echavemann) [inline poetry.lock:1]: versins still dont match... maybe just drop the diff of this

File: src/pysrc/signal/example_feature_generator.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        snapshots: dict[str, SnapshotMessage],
+        trades: dict[str, list[TradeMessage]],
+    ) -> dict[Asset, dict[str, list[float]]]:
+        output = {}
+        for asset in self.assets:
+            feedcode = asset.name
+            asset_trades = trades.get(feedcode, [])
+            features = self.compute_ohlc(asset_trades)
+            output[asset] = {"features": features}
+        # ["open", "high", "low", "close"]
```
Comment (echavemann) [inline src/pysrc/signal/example_feature_generator.py:36]: del

File: src/pysrc/test/unit/signal/test_example_feature_generator.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import pytest
+from pysrc.signal.example_feature_generator import ExampleFeatureGenerator
+from pysrc.adapters.messages import TradeMessage, SnapshotMessage
+from pysrc.util.types import Asset, OrderSide, Market
+
+
+def create_trade(
```
Comment (echavemann) [inline src/pysrc/test/unit/signal/test_example_feature_generator.py:7]: why? I just meant use a variable to store the big list of trades

Comment (echavemann) [reply src/pysrc/test/unit/signal/test_example_feature_generator.py:7]: this fn is very whatever, id delete

Comment (echavemann) [reply src/pysrc/test/unit/signal/test_example_feature_generator.py:7]: along with the fixtures

File: poetry.lock (hunk 2 above)
Comment (minnce) [reply poetry.lock]: fixed in latest I believe

**Review by echavemann — APPROVED** [review]

**Pull request merged**